api/supportbundle.go

```diff
@@ -0,0 +1,103 @@
+package api
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+
+	"k8s.io/apimachinery/pkg/labels"
+
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta1"
+	"github.com/longhorn/longhorn-manager/types"
+	"github.com/longhorn/longhorn-manager/util"
+)
+
+func (s *Server) SupportBundleDownload(rw http.ResponseWriter, r *http.Request) error {
+	bundleName := mux.Vars(r)["bundleName"]
+
+	retainSb := false
+	if retain, err := strconv.ParseBool(r.URL.Query().Get("retain")); err == nil {
+		retainSb = retain
+	}
+
+	sb, err := s.m.GetSupportBundle(bundleName)
+	if err != nil {
+		util.ResponseError(rw, http.StatusBadRequest, errors.Wrap(err, "fail to get support bundle resource"))
+		return err
+	}
+
+	if sb.Status.State != types.SupportBundleStateReady || sb.Status.FileName == "" || sb.Status.FileSize == 0 {
+		util.ResponseError(rw, http.StatusBadRequest, errors.New("support bundle is not ready"))
+		return err
+	}
+
+	managerPodIP, err := s.getManagerPodIP(sb)
+	if err != nil {
+		util.ResponseError(rw, http.StatusBadRequest, errors.Wrap(err, "fail to get support bundle manager pod IP"))
+		return err
+	}
+
+	url := fmt.Sprintf("http://%s:8080/bundle", managerPodIP)
+	req, err := http.NewRequestWithContext(r.Context(), http.MethodGet, url, nil)
+	if err != nil {
+		util.ResponseError(rw, http.StatusInternalServerError, err)
+		return err
+	}
+	resp, err := s.httpClient.Do(req)
```

Comment:
We cannot download to the mgr's local directory. Need to make it possible for the UI to download the zip file.